crates/sui-core/src/consensus_adapter.rs

```diff
@@ -886,6 +886,15 @@ impl ConsensusAdapter {
                 vec![]
             };
 
+            let checkpoint_sequence_number = if let SequencedConsensusTransactionKey::External(
+                ConsensusTransactionKey::CheckpointSignature(_, checkpoint_sequence_number),
+            ) = transaction_key
+            {
+                Some(checkpoint_sequence_number)
+            } else {
+                None
```

Comment:
instead of doing Some/None, you could do:

let cktp_future = if ... {
    Either::Left(epoch_store.synced_checkpoint_notify(checkpoint_sequence_number))
} else {
    Either::Right(std::future::pending())
}

and then below you can just do `processed = ckpt_future` - if there is no checkpoint sequence number then that arm of the select will simply never resolve

Reply:
Didn't cross my mind the use of `Either` here - thanks @mystenmark will try this!